ts/pages/staking/staking_pool.tsx

```diff
@@ -240,70 +248,85 @@ const tradingPairs = [
 export const StakingPool: React.FC<StakingPoolProps & RouteChildrenProps> = props => {
     const { poolId } = useParams();
 
+    const networkId = useSelector((state: State) => state.networkId);
+    const apiClient = useAPIClient(networkId);
+    const [stakingPool, setStakingPool] = useState<PoolWithHistoricalStats | undefined>(undefined);
+
+    useEffect(() => {
+        apiClient
+            .getStakingPoolByIdAsync(poolId)
+            .then(res => setStakingPool(res.stakingPool))
+            .catch(e => logUtils.warn(e));
+    }, [poolId, setStakingPool, apiClient]);
```

Comment:
Wouldn't `apiClient` have a new reference on each render given that we just instantiate and return a new instance in the hook? Or is this magically handled by React?

```suggestion
    }, [poolId, setStakingPool, apiClient.account]);
```

Reply:
no the useApiClient hook holds on to the same reference of apiClient unless the network Id changes

Reply:
(if you're curious

https://github.com/0xProject/website/blob/development-staking/ts/hooks/use_api_client.ts#L6
this ref holds on to the reference between renders

and will only update when networkId changes -- https://github.com/0xProject/website/blob/development-staking/ts/hooks/use_api_client.ts#L8-L10
)

Reply:
(I had refactored it a little bit ago to be more 'hook'ey)